### PR TITLE
Add library documentation

### PR DESCRIPTION
**Pull request by swelham**
Adds full documentation for Faktory Worker.

Tasks
* [x] Update the README with
  * Getting started
  * Contributing
  * Developing
* [x] Add a docs page about configuring Faktory Worker
* [x] Document all public modules, functions and types
* [x] Bump version to `1.0.0`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,114 @@
...
-A work in progress Faktory worker for Elixir.
+A worker client for [Faktory](https://github.com/contribsys/faktory).
+
+## Documentation
+
+Faktory Worker documentation is available at [https://hexdocs.pm/faktory_worker](https://hexdocs.pm/faktory_worker).
+
+## Getting up and running
+
+To get started with Faktory Worker first add the dependancy to your `mix.exs` file.
```
Comment (john-griffin) [inline README.md]:
```suggestion
To get started with Faktory Worker first add the dependency to your `mix.exs` file.
```

File: README.md [hunk 2]
```diff
@@ -1,3 +1,114 @@
...
+
+Now you can start sending jobs to faktory and they will be automatically picked up by the worker.
+
+```elixir
+:ok = MyApp.SomeWorker.perform_async("hey there!")
+```
+
+## Sending multiple job arguments
+
+It's possible to send more that one argument to Faktory by passing a list of data to `perform_async/1`. Picking up from the example in the [Getting up and running](#getting-up-and-running) section we can modify our worker by adding another `perform` function with an artity that matches the number of job arguments we expect to send to Faktory.
```
Comment (john-griffin) [inline README.md]:
```suggestion
It's possible to send more than one argument to Faktory by passing a list of data to `perform_async/1`. Picking up from the example in the [Getting up and running](#getting-up-and-running) section we can modify our worker by adding another `perform` function with an arity that matches the number of job arguments we expect to send to Faktory.
```

File: README.md [hunk 3]
```diff
@@ -1,3 +1,114 @@
...
+
+The full list of configuration options are available in the [Configuration](configuration.html) documentation.
+
+## Contributing
+
+We always appreciate contributions whether they are testing, reporting issues, feedback or submitting PRs. If you would like to work on Faktory Worker please follow the [Developing](#developing) section for details on how to get setup for developing and running the test suite.
+
+## Developing
+
+Faktory Worker includes a docker compose file that provisons all of the Faktory instances required to run the test suite.
```
Comment (john-griffin) [inline README.md]:
```suggestion
Faktory Worker includes a docker compose file that provisions all of the Faktory instances required to run the test suite.
```

File: README.md [hunk 4]
```diff
@@ -1,3 +1,114 @@
...
+
+    Supervisor.start_link(children,
+      strategy: :one_for_one,
+      name: MyApp.Supervisor
+    )
+  end
+end
+```
+
+Make sure you have configured your application to start in your `mix.exs` file.
```
[the remark this answers is not in the document]
Comment (swelham) [reply README.md]: Nope this isn't setup when using `mix new` and has to be done manually. The default config is
```elixir
def application do
  [
    extra_applications: [:logger]
  ]
end
```
The application is configured by default when creating a new phoenix app though.

**Comment by swelham** [comment]
Thanks for the typo finds. I have since installed a spellchecker!

**Pull request merged**